### PR TITLE
feat: check and apply markdown for smart picker output

### PR DESCRIPTION
**Pull request by luka-nextcloud**
### 📝 Summary

* Resolves: #4784

Always consider smart picker output as markdown.

### 🚧 TODO

- [ ] ...

### 🏁 Checklist

- [ ] Code is properly formatted (`npm run lint` / `npm run stylelint` / `composer run cs:check`)
- [ ] [Sign-off message](https://probot.github.io/apps/dco/) is added to all commits
- [ ] [Tests](https://github.com/nextcloud/text#-testing-the-app) (unit, integration and/or end-to-end) passing and the changes are covered with tests
- [ ] Documentation ([README](https://github.com/nextcloud/text/blob/main/README.md) or [documentation](https://github.com/nextcloud/documentation/blob/master/admin_manual/configuration_server/text_configuration.rst#L2)) has been updated or is not required

### CODE REVIEW

File: src/components/Suggestion/LinkPicker/suggestions.js [hunk 1]
```diff
@@ -34,6 +35,31 @@ const filterOut = (e) => {
 
 const important = ['task-list', 'table']
 
+const hasMarkdownSyntax = (content) => {
+	// Regular expressions for common Markdown patterns
+	const markdownPatterns = [
+		/\*\*.*?\*\*/, // Bold: **text**
+		/\*.*?\*/, // Italics: *text*
+		/\[.*?\(.*?\)/, // Links: [text](url)
+		/^#\s.*$/, // Headings: # text
```
Comment (juliushaertl) [inline src/components/Suggestion/LinkPicker/suggestions.js:44]: Would be good to also cover headings of other levels h2-h6

File: src/components/Suggestion/LinkPicker/suggestions.js [hunk 2]
```diff
@@ -34,6 +35,31 @@ const filterOut = (e) => {
...
+		/- .*/, // Unordered list: - item
```
Comment (juliushaertl) [inline src/components/Suggestion/LinkPicker/suggestions.js:45]: Markdown also supports `+` and `*` as list entries

File: src/components/Suggestion/LinkPicker/suggestions.js [hunk 3]
```diff
@@ -34,6 +35,31 @@ const filterOut = (e) => {
...
+		/\d\..*/, // Ordered list: 1. item
```
Comment (juliushaertl) [inline src/components/Suggestion/LinkPicker/suggestions.js:46]:
```suggestion
		/^\s\d\..*/m, // Ordered list: 1. item
```

File: src/components/Suggestion/LinkPicker/suggestions.js [hunk 4]
```diff
@@ -34,6 +35,31 @@ const filterOut = (e) => {
...
+		/^>\s.*/, // Blockquote: > text
```
Comment (juliushaertl) [inline src/components/Suggestion/LinkPicker/suggestions.js:47]:
```suggestion
		/^>+\s.*/, // Blockquote: > text
```

File: src/components/Suggestion/LinkPicker/suggestions.js (hunk 2 above)
Comment (juliushaertl) [inline src/components/Suggestion/LinkPicker/suggestions.js:45]:
```suggestion
		/^\s*[+-*]\s.*/m, // Unordered list: - item
```

**Review by juliushaertl — CHANGES_REQUESTED** [review]
A few suggestions for detecting the markdown syntax (I know we can't cover everything there), but otherwise this seems fine and works well 👍

**Comment by mejo-** [comment]
Maybe I'm missing something here, but why do we have to detect markdown manually at all, instead of just doing `isValidMarkdown()` and always sending the content through `markdownit.render()` if that succeeds?

**Comment by juliushaertl** [comment]
Hm,  unsure about that. It might be fine to just always pass it to markdown it indeed.

**Comment by luka-nextcloud** [comment]
I have tried this way and faced an issue. The markdown-it render function always adds some newline character to content. So, If input value is an url, the inserted content will be a normal string instead of a link (enclose it in an angle brackets).

**Review by juliushaertl — APPROVED** [review]

**Pull request merged**